src/extension/wps/src/main/java/au/org/emii/gogoduck/worker/GoGoDuck.java

```diff
@@ -409,25 +405,37 @@ public static int execute(List<String> command) throws Exception {
         builder.redirectErrorStream(true);
 
         final Process process = builder.start();
+        StringBuilder bashLog = new StringBuilder();
 
         try (
             InputStream is = process.getInputStream();
             InputStreamReader isr = new InputStreamReader(is);
             BufferedReader br = new BufferedReader(isr)
         ) {
             String line;
+
             while ((line = br.readLine()) != null) {
                 logger.info(line);
+                bashLog.append(line);
             }
 
             process.waitFor();
 
+            if (process.exitValue() != 0) {
```

Comment:
I'm a bit concerned about the handling here of process. In my testing, the exception is thrown and bubbled up but by the time we have an exit value for the process, the damage has already been done and the file has been created (and therefore returned). So we get an exception in our catalina log. So, my suggestion would be to consider setting the processListener.isCancelled(true) whenever we throw a GGD exception. Needs testing and validation of course.